### PR TITLE
fix: stop active run after AskUserQuestion

### LINKED ISSUE #606: AskUserQuestion leaves the active Agent Run running after sending the card
## Description / 问题描述

After `feishu_ask_user_question` successfully creates and sends an interactive card, the tool stores the pending question and returns `status: "pending"`. However, the current Agent Run remains active.

The tool description asks the model to wait, but that is only a prompt-level instruction. The model can continue the same turn and infer or generate an answer even though the user has not submitted the card.

The current implementation returns directly after storing pending state and has no session-scoped run termination:
https://github.com/larksuite/openclaw-lark/blob/dde0be3680d6fd5443cab426c8f4b3216266346a/src/tools/ask-user-question.ts#L994-L1083

This was observed with plugin `2026.5.12`. Source review confirms the same control-flow gap is still present on current `main` (`dde0be3`, published as `2026.7.16`).

## Error Message / 错误信息

No exception is emitted. The tool returns a normal pending result, but the active Agent Run can continue producing assistant output before any card submission.

## Expected Behavior / 期望行为

Once all three steps have succeeded:

1. the card entity is created,
2. the card is delivered, and
3. the pending question is stored,

the current session's Agent Run should stop after the pending tool result is returned. The user submission callback should then inject the answer and start the follow-up run as it already does.

Failures before pending state is established must not stop the run, and stopping one session must not affect any other session.

## openclaw Version / openclaw 版本

2026.5.3 (observed); current plugin peer range is `>=2026.5.4`

## openclaw-lark (this plugin) Version / openclaw-lark（本插件）版本

2026.5.12 (observed); current main / 2026.7.16 remains affected by source inspection

## OS Version / 操作系统版本

Linux container

## Steps to Reproduce / 复现步骤

1. Start a Feishu conversation backed by an OpenClaw Agent.
2. Have the Agent call `feishu_ask_user_question`.
3. Confirm that the interactive question card is delivered.
4. Do not select or submit any answer.
5. Observe that the same Agent Run is still allowed to continue and may produce another assistant response.

A robust fix is to register the tool with its invocation context, require the current `sessionId`, and schedule a session-scoped `abortAgentHarnessRun(sessionId)` only after card delivery and pending-state storage have succeeded.

### PR DESCRIPTION
**Pull request by BenGuanRan**
Closes #606.

## Summary

- register `feishu_ask_user_question` as a context-aware tool so it receives the active `sessionId`
- return the pending tool result, then stop only that session's Agent Run on the next event-loop turn
- schedule the abort only after card creation, card delivery, and pending-question storage succeed
- fail before creating a card when no active session is available
- align the development OpenClaw version with the existing `>=2026.5.4` peer floor that provides the agent-harness abort API

The existing Feishu `cards` runtime capability is unchanged. The small type cast in `actions.ts` only bridges the narrower OpenClaw 2026.5.x declaration.

## Why

Returning `status: "pending"` and asking the model to wait is advisory. Without stopping the active run, the same turn can continue and produce an answer before the user submits the interactive card.

The callback path remains unchanged: a real card submission is still injected as a follow-up message and handled by a new run.

## Validation

- regression test demonstrated the old behavior: expected one session abort, received zero
- `vitest run --pool=threads --maxWorkers=1`: 53 files, 445 tests passed
- `tsc --noEmit`: passed
- `eslint src/ index.ts secret-contract-api.ts`: passed with 0 errors (26 existing warnings)
- `tsdown`: passed
- pnpm 10.32.1 frozen-lockfile install: passed

### CODE REVIEW

**Comment by BenGuanRan** [comment]
@Nirvana-Jie @HanShaoshuai-k could you please review this fix?

It preserves the non-blocking synthetic-message flow introduced in #253, while closing the control-flow gap described in #606: after the question card is delivered and pending state is stored, only the current session's Agent Run is stopped. Card creation/delivery failures do not trigger an abort, and the submission callback path is unchanged.

The regression test demonstrated zero aborts on the old implementation; the updated branch passes all 445 tests, typecheck, lint, and build.

**Comment by BenGuanRan** [comment]
@evandance @liangshuo-1, apologies for the direct ping. If you have the bandwidth, could you please help review this PR?

It addresses #606 and is intentionally scoped to stop only the active Agent Run after the question card has been delivered and the pending state has been stored. The callback path remains unchanged, and all local checks are green (445 tests, typecheck, lint, and build).

I understand you may be busy, so any feedback whenever convenient would be greatly appreciated. I’ll respond to any requested changes promptly. Thank you very much for your time and for maintaining the project.

**Review by Nirvana-Jie — COMMENTED** [review]
Overall, the approach aligns with #606: the run is scoped to the active session and aborted on the next event-loop turn, while the callback path remains unchanged. Local validation passed (445 tests, typecheck, lint, formatting, and build). There is one delivery-success edge case to address before relying on the abort.

File: src/tools/ask-user-question.ts [hunk 1]
```diff
@@ -1073,16 +1080,28 @@ export function registerAskUserQuestionTool(api: OpenClawPluginApi): void {
...
+      const result = formatToolResult({
         status: 'pending',
         questionId,
         message:
           'Question card sent to the user. Their answers will arrive as a follow-up message ' +
           'in this conversation. Do NOT call this tool again for the same question — just wait ' +
           'for the response message.',
       });
+
+      setImmediate(() => {
```
Comment (Nirvana-Jie) [inline src/tools/ask-user-question.ts:1095]: **[P1] Verify business-level delivery before aborting the run**

The precondition for scheduling this abort is only that `sendCardByCardId()` fulfilled. That helper does not validate `response.code` and converts a missing `message_id` to an empty string. As a result, a resolved business failure or a response without a message ID can still store pending state and abort the run even though the user received no card.

Please make `sendCardByCardId` throw on a non-zero response code or a missing `message_id`, and require a non-empty returned `messageId` before storing pending state or scheduling the abort. Add Vitest cases for a resolved non-zero business response and a resolved response without `message_id`; both should assert that the abort is not called. The current success helper uses `mockResolvedValue(undefined)`, so it cannot catch this branch.

**Comment by Nirvana-Jie** [comment]
@mazhe-nerd  Also PTAL～

File: src/tools/ask-user-question.ts (hunk 1 above)
Comment (BenGuanRan) [reply src/tools/ask-user-question.ts:1095]: Confirmed — this is a real issue. `@larksuiteoapi/node-sdk` resolves HTTP-200 responses with a non-zero business code, so fulfillment alone is not a delivery guarantee.

I have implemented the fix locally:
- `sendCardByCardId()` now throws on a non-zero `code` or a missing/empty `message_id` for both reply and create paths.
- `AskUserQuestion` requires a non-empty returned `messageId` before storing pending state or scheduling the run abort.
- Added regression coverage for a resolved business failure and a resolved response without `message_id`; neither path aborts the run.

Verification: focused tests 28/28, full suite 448/448, typecheck/build/format pass, lint 0 errors (26 existing warnings). The branch update has not been pushed yet.

Comment (BenGuanRan) [reply src/tools/ask-user-question.ts:1095]: request re-review please~

Comment (BenGuanRan) [reply src/tools/ask-user-question.ts:1095]: The fix described above is now pushed in 52c5402 (was previously local-only, sorry for the confusion):

- `sendCardByCardId` throws via `requireSentMessage` on a non-zero IM business `code` or a missing/empty `message_id`, on both the reply and create paths.
- `AskUserQuestion` requires a non-empty returned `messageId` before storing pending state or scheduling the run abort.
- Added the two requested regression cases (resolved business failure `code=230001`, and resolved response without `message_id`), both asserting the abort is not called, plus fetch-level tests for `sendCardByCardId` itself (`tests/cardkit-send.test.ts`).
- The success helper now resolves a real `{ messageId, chatId }` instead of `undefined`.

Verification: focused tests 28/28 pass, `tsc --noEmit` clean, eslint 0 errors on the touched files.

**Comment by BenGuanRan** [comment]
@mazhe-nerd could you please help review this PR?

**Comment by BenGuanRan** [comment]
@Nirvana-Jie @evandance, gentle follow-up: the delivery-success edge case raised in review is fixed in `52c5402`, with regression coverage for a non-zero IM business code and a missing `message_id`. The review thread is now awaiting re-review. When you have bandwidth, could you please take another look? Thank you.
